src/build/app/settings.rs

```diff
@@ -689,6 +692,83 @@ pub enum AppSettings {
     /// [`subcommands`]: crate::App::subcommand()
     DeriveDisplayOrder,
 
+    /// Specifies that the parser should match first argument as [`subcommands`],
+    /// treat any matches as if the subcommand had been passed,
+    /// and print arguments like the [`subcommands`] are the top-level commands,
+    /// but if none match parse as usual.
+    ///
+    /// This is desired behaviour for "multicall" executables such as busybox
+    /// which dispatch to which applet to run
+    /// based on the name the executable has been linked as.
+    ///
+    /// # Panics
+    ///
+    /// If this option is combined with [`NoBinaryName`] then
+    /// [`try_get_matches_from_mut`] will [`panic!`].
+    ///
+    /// # Examples
+    ///
+    /// It does not make sense to combine NoBinaryName with Multicall
+    /// so it will panic.
+    ///
+    /// ```should_panic
+    /// # use clap::{App, AppSettings};
+    /// let _ = App::new("busybox")
+    ///     .setting(AppSettings::NoBinaryName)
+    ///     .setting(AppSettings::Multicall)
+    ///     .subcommand(App::new("true"))
+    ///     .subcommand(App::new("false"))
+    ///     .get_matches_from(&["true"]);
+    /// ```
```

Comment:
I'm unsure if we need a panic example.  If we do, we should not probably make it the front and center example but more buried so people can focus on the cases they are more likely to run into

Reply:
It made more sense when the function itself was responsible for the panic, documenting that it's incompatible should be enough.